### PR TITLE
Use existing standards to define kex calculations

### PR DESCRIPTION
**Pull request by tomato42**
Use existing standards to describe key exchange

plus a small fix for address

### CODE REVIEW

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 1]
```diff
@@ -196,32 +196,21 @@
...
-                <t>Converts the point q_C to the Q_C octet string by
-                    concatenation of value 0x04 and big-endian representation
-                    of the x coordinate and then y coordinate. The coordinate
-                    coversion MUST preserve leading zero octets. Thus for
-                    nistp521 curve the encoded x coordinate will always have a
-                    length of 66 octets while the Q_C representation will be
-                    133 octets long. This is the uncompressed representation
-                    specified in Section 4.3.6 of
-                    <xref target="ANSI-X9-62-2005"/>.</t>
+                <t>Following algorithm in Section 3.2.1 of
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: the algorithm

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 2]
```diff
@@ -196,32 +196,21 @@
...
+                    <xref target="SEC1v2"/> to create a private key d_C and
+                    public key q_C using the curve paramaters associated
+                    with the selected key exchange method.</t>
+                <t>Coverting the resulting point q_C to the Q_C octet string
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: Converting

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 3]
```diff
@@ -284,28 +273,16 @@
                 or 56 octets for curve448. If the value does not have matching
                 length the key exchange MUST fail.</t>
             <t>In case of key exchanges that use NIST curves, the server MUST
-                check if the first octet of the Q_C is equal to 0x04.
-                If the octet has different value the key exchange MUST fail.
+                convert the Q_C to a point representation q_C and verify
+                that the Q_C used uncompressed point representation
+                using algorithm in Section 2.3.4 of <xref target="SEC1v2"/>.
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: the algorithm

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 4]
```diff
@@ -284,28 +273,16 @@
...
-                (satisfies the curve equation). See
-                <xref target="ANSI-X9-62-2005"/> for detailed description and
-                rationale.</t>
-            <t>For curve25519, the server verifies that the the high-order
-                bit of the last octet is not set - this prevents distinguishing
-                attacks between implementations that use Montgomery ladder
-                implementation of the curve and ones that use generic
-                elliptic-curve libraries. If the bit is set, the key exchange
-                SHOULD fail. For curve448 any bit can be set.</t>
+            <t>For NIST curves, the server performs a partial public key
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: the partial

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 5]
```diff
@@ -345,15 +322,14 @@
...
-              <t>For NIST curves, the peers perform point multiplication using
-                  d_U and q_V to get point P.</t>
-              <t>For NIST curves, peers verify that P is not a point at
-                  infinity. If P is a point at infinity, the key exchange
-                  MUST fail.</t>
-              <t>For NIST curves, the shared secret is the zero-padded
-                  big-endian representation of the x coordinate of P -
-                  32 octets long for nistp256, 48 octets for nistp384 and 64
-                  octets long for nistp521.</t>
+              <t>For NIST curves, the peers calculate shared secret field
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: the shared secret

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 6]
```diff
@@ -345,15 +322,14 @@
...
+                  element using d_U and q_V using algorithms in Section 3.3.1
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: the algorithm

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 7]
```diff
@@ -345,15 +322,14 @@
...
+                  of <xref target="SEC1v2"/>. If the output of that algorithm
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: If the result is "invalid" ...

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 8]
```diff
@@ -345,15 +322,14 @@
...
+                  is "invalid" (the shared secret point is a point
+                  at infinity), the key exchange MUST fail.</t>
+              <t>For NIST curves, the shared secret is the octet string
+                  representation of the field element using algorithm described
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: the algorithm

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 9]
```diff
@@ -345,15 +322,14 @@
...
+                  in Section 2.3.5 in <xref target="SEC1v2"/>.</t>
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: 2.3.5 of

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 10]
```diff
@@ -638,21 +614,21 @@
       <!--ECC specifications-->
 
       <reference
-          anchor="ANSI-X9-62-2005">
+          anchor="SEC1v2">
           <front>
-              <title>Public Key Cryptography for the Financial Services
```
Comment (simo5) [inline draft-ietf-curdle-gss-keyex-sha2.xml:643]: Why change this ?

**Review by simo5 — CHANGES_REQUESTED** [review]
Mostly minor nitpicks

File: draft-ietf-curdle-gss-keyex-sha2.xml (hunk 10 above)
Comment (tomato42) [reply draft-ietf-curdle-gss-keyex-sha2.xml:643]: Because I don't have it on hand and the SEC1 is freely available

**Comment by tomato42** [comment]
@simo5 updated

**Review by simo5 — APPROVED** [review]
LGTM

**Pull request merged**